pkg/agent/controller/egress/egress_controller.go

```diff
@@ -745,6 +768,9 @@ func (c *EgressController) updateEgressStatus(egress *crdv1b1.Egress, egressIP s
 					Message:            fmt.Sprintf("Failed to assign the IP to EgressNode: %v", scheduleErr),
 				},
 			}
+			if c.record != nil {
```

Comment:
No need to add a special check for unit test, it can be faked via `record.NewFakeRecorder`